usage/src/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java

```diff
@@ -0,0 +1,444 @@
+
+    public QuotaAlertManagerImpl() {
+        super();
+    }
+
+    private void mergeConfigs(Map<String, String> dbParams, Map<String, Object> xmlParams) {
+        for (Map.Entry<String, Object> param : xmlParams.entrySet()) {
+            dbParams.put(param.getKey(), (String) param.getValue());
+        }
+    }
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        super.configure(name, params);
+
+        Map<String, String> configs = _configDao.getConfiguration(params);
+
+        if (params != null) {
+            mergeConfigs(configs, params);
+        }
+
+        final String smtpHost = configs.get(QuotaConfig.QuotaSmtpHost.key());
+        int smtpPort = NumbersUtil.parseInt(configs.get(QuotaConfig.QuotaSmtpPort.key()), 25);
+        String useAuthStr = configs.get(QuotaConfig.QuotaSmtpAuthType.key());
+        boolean useAuth = ((useAuthStr != null) && Boolean.parseBoolean(useAuthStr));
+        String smtpUsername = configs.get(QuotaConfig.QuotaSmtpUser.key());
+        String smtpPassword = configs.get(QuotaConfig.QuotaSmtpPassword.key());
+        String emailSender = configs.get(QuotaConfig.QuotaSmtpSender.key());
+        _lockAccountEnforcement = "true".equalsIgnoreCase(configs.get(QuotaConfig.QuotaEnableEnforcement.key()));
+        _emailQuotaAlert = new EmailQuotaAlert(smtpHost, smtpPort, useAuth, smtpUsername, smtpPassword, emailSender, _smtpDebug);
+
+        return true;
+    }
+
+    @Override
+    public boolean start() {
+        if (s_logger.isInfoEnabled()) {
+            s_logger.info("Starting Alert Manager");
+        }
+        _pid = Integer.parseInt(System.getProperty("pid"));
+        return true;
+    }
+
+    @Override
+    public boolean stop() {
+        if (s_logger.isInfoEnabled()) {
+            s_logger.info("Stopping Alert Manager");
+        }
+        return true;
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public void sendMonthlyStatement(Date now) {
+        Calendar aCalendar = Calendar.getInstance();
+        aCalendar.add(Calendar.MONTH, -1);
+        aCalendar.set(Calendar.DATE, 1);
+        aCalendar.set(Calendar.HOUR, 0);
+        aCalendar.set(Calendar.MINUTE, 0);
+        aCalendar.set(Calendar.SECOND, 0);
+        Date firstDateOfPreviousMonth = aCalendar.getTime();
+        aCalendar.set(Calendar.DATE, aCalendar.getActualMaximum(Calendar.DAY_OF_MONTH) + 1);
+        Date lastDateOfPreviousMonth = aCalendar.getTime(); // actually the first day of this month
+
+        List<DeferredQuotaEmail> deferredQuotaEmailList = new ArrayList<DeferredQuotaEmail>();
+        for (final QuotaAccountVO quotaAccount : _quotaAcc.listAll()) {
+            if (quotaAccount.getQuotaBalance() == null) {
+                continue; // no quota usage for this account ever, ignore
+            }
+            Date lastStatementDate = quotaAccount.getLastStatementDate();
+            if (now.getDate() < 6) {
```

Comment:
@bhaisaab  got the right meaning. again a hard coding. should be put as a global config variable.